lms/djangoapps/certificates/models.py

```diff
@@ -1264,3 +1264,43 @@ class Meta:
 
     def __str__(self):
         return str(self.arguments)
+
+@python_2_unicode_compatible
+class CertificateDateOverride(TimeStampedModel):
+    """
+    Model to manually override a given certificate date with the given date.
+
+    .. no_pii:
+    """
+    generated_certificate = models.OneToOneField(
+        GeneratedCertificate,
+        on_delete=models.CASCADE,
+        primary_key=True,
+        related_name='date_override',
+        help_text="The id of the Generated Certificate to override",
+    )
+    date = models.DateField(
+        help_text="The date to display on the certificate",
+    )
+    reason = models.TextField(
+        help_text="The reason why you are overriding the certificate date (Update this when you add OR edit the date.)",
+    )
+    overridden_by = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
```

Comment:
This seems wrong 🤔 

The `on_delete` argument defines what to do *when the object referenced by the ForeignKey* is deleted. What do we want to happen to this `CertificateDateOverride` object when the associated `User` is deleted?

`models.CASCADE` (the value I see most places) means that this `CertificateDateOverride` would be deleted, too! Even though a User will probably never be deleted, I don’t think we want this.

Docs (with some alternatives): https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.ForeignKey.on_delete

Reply:
User's never get deleted, so I think leaving CASCADE is fine.

Reply:
From discussion at Standup: Hm. Didn’t settle on an alternative: 
- `SET_NULL` — would make sense, but requires us to allow `Null`, which we **don’t** want the rest of the time.
- `SET_DEFAULT` — ... to what? do we have some sort of default admin user?
- `PROTECT` — would prevent deleting the `User`. Maybe??? Why/why not?
- `DO_NOTHING` — 😄 Why/why not?

Reply:
I think Cascade makes the most sense here - none of the other values seem appropriate. And if you **really did** delete a user, you'd want to delete this too, yeah?

Reply:
No, we would **not** want this record to be deleted if the `user` was deleted. See note in initial comment:
> Remember, this is the admin user that made the change; not the user that the associated certificate belongs to.

For example: if an edX employee with admin privileges rightfully added a bunch of these overrides, and then left the company and we decided to delete their user account for some reason, we would probably not want to delete all the rightful overrides they made.

Reply:
Ah yeah my bad. Then I'd think anything but PROTECT or CASCADE would make sense. SET_NULL or DO_NOTHING if you are willing to guard your uses of `.user`. And SET_DEFAULT if you want to create a 'DeletedUser` sentinel object or similar (I don't know if we have one already or not).

All of those are a little annoying if we truly never delete any user (which I think we don't). So maybe DO_NOTHING and just hope it never happens (I'm guessing if it did happen, referencing `.user` would throw some error). With maybe a little comment explaining why you chose that non-typical option.

Reply:
After discussion: Going with `DO_NOTHING`!